Flaky test in nightlies: target distribution data check

I saw the following fail in [the linux nightlies yesterday](https://github.com/alteryx/evalml/runs/3228179479)
```
FAILED evalml/tests/data_checks_tests/test_target_distribution_data_check.py::test_target_distribution_data_check_unsupported_target_type[double]
...
        if target_type in ["integer", "double"]:
>           assert target_dist_check.validate(X, y) == {
                "warnings": [],
                "errors": [],
                "actions": [],
            }
E           AssertionError: assert {'actions': [...rning', ...}]} == {'actions': [...warnings': []}
E             Omitting 1 identical items, use -vv to show
E             Differing items:
E             {'actions': [{'code': 'TRANSFORM_TARGET', 'metadata': {'column': None, 'is_target': True, 'transformation_strategy': 'lognormal'}}]} != {'actions': []}
E             {'warnings': [{'code': 'TARGET_LOGNORMAL_DISTRIBUTION', 'data_check_name': 'TargetDistributionDataCheck', 'details': {'shapiro-statistic/pvalue': '0.969/0.018'}, 'level': 'warning', ...}]} != {'warnings': []}
E             Use -v to get the full diff

evalml/tests/data_checks_tests/test_target_distribution_data_check.py:59: AssertionError
```

Tagging @ParthivNaresh since I think he touched this code recently and may have an idea on the fix

--- a/docs/source/release_notes.rst
+++ b/docs/source/release_notes.rst
@@ -7,6 +7,7 @@ Release Notes
         * Patched bug in ``xgboost`` estimators where predicting on a feature matrix of only booleans would throw an exception. :pr:`2602`
     * Fixes
         * Updated ``get_best_sampler_for_data`` to consider all non-numeric datatypes as categorical for SMOTE :pr:`2590`
+        * Fixed inconsistent test results from `TargetDistributionDataCheck` :pr:``
     * Changes
         * Renamed SMOTE samplers to SMOTE oversampler :pr:`2595`
     * Documentation Changes

--- a/docs/source/release_notes.rst
+++ b/docs/source/release_notes.rst
@@ -7,7 +7,7 @@ Release Notes
         * Patched bug in ``xgboost`` estimators where predicting on a feature matrix of only booleans would throw an exception. :pr:`2602`
     * Fixes
         * Updated ``get_best_sampler_for_data`` to consider all non-numeric datatypes as categorical for SMOTE :pr:`2590`
-        * Fixed inconsistent test results from `TargetDistributionDataCheck` :pr:``
+        * Fixed inconsistent test results from `TargetDistributionDataCheck` :pr:`2608`
     * Changes
         * Renamed SMOTE samplers to SMOTE oversampler :pr:`2595`
     * Documentation Changes

--- a/evalml/data_checks/target_distribution_data_check.py
+++ b/evalml/data_checks/target_distribution_data_check.py
@@ -30,14 +30,14 @@ class TargetDistributionDataCheck(DataCheck):
         Example:
             >>> from scipy.stats import lognorm
             >>> X = None
-            >>> y = lognorm.rvs(0.1, size=1000)
-            >>> target_checdk = TargetDistributionDataCheck()
+            >>> y = [0.946, 0.972, 1.154, 0.954, 0.969, 1.222, 1.038, 0.999, 0.973, 0.897]
+            >>> target_check = TargetDistributionDataCheck()
             >>> assert target_check.validate(X, y) == {"errors": [],\
                                                        "warnings": [{"message": "Target may have a lognormal distribution.",\
                                                                     "data_check_name": "TargetDistributionDataCheck",\
                                                                     "level": "warning",\
                                                                     "code": "TARGET_LOGNORMAL_DISTRIBUTION",\
-                                                                    "details": {"shapiro-statistic/pvalue": '0.993/0.0'}}],\
+                                                                    "details": {"shapiro-statistic/pvalue": '0.84/0.045'}}],\
                                                         "actions": [{'code': 'TRANSFORM_TARGET', 'metadata': {'column': None, 'transformation_strategy': 'lognormal', 'is_target': True}}]}
         """
         results = {"warnings": [], "errors": [], "actions": []}
@@ -78,7 +78,7 @@ class TargetDistributionDataCheck(DataCheck):
         if shapiro(y).pvalue >= 0.05:
             return results
 
-        y_new = y
+        y_new = round(y, 6)
         if any(y <= 0):
             y_new = y + abs(y.min()) + 1
 
@@ -97,7 +97,7 @@ class TargetDistributionDataCheck(DataCheck):
 
         if log_detected:
             details = {
-                "shapiro-statistic/pvalue": f"{round(shapiro_test_og.statistic, 3)}/{round(shapiro_test_og.pvalue, 3)}"
+                "shapiro-statistic/pvalue": f"{round(shapiro_test_og.statistic, 2)}/{round(shapiro_test_og.pvalue, 3)}"
             }
             results["warnings"].append(
                 DataCheckWarning(

--- a/evalml/tests/data_checks_tests/test_target_distribution_data_check.py
+++ b/evalml/tests/data_checks_tests/test_target_distribution_data_check.py
@@ -37,19 +37,17 @@ def test_target_distribution_data_check_no_y(X_y_regression):
 
 
 @pytest.mark.parametrize("target_type", ["boolean", "categorical", "integer", "double"])
-def test_target_distribution_data_check_unsupported_target_type(
-    target_type, X_y_regression
-):
-    X, y = X_y_regression
+def test_target_distribution_data_check_unsupported_target_type(target_type):
+    X = pd.DataFrame(range(5))
 
     if target_type == "boolean":
-        y = pd.Series([True, False] * 50)
+        y = pd.Series([True, False] * 5)
     elif target_type == "categorical":
-        y = pd.Series(["One", "Two", "Three", "Four"] * 25)
+        y = pd.Series(["One", "Two", "Three", "Four", "Five"] * 2)
     elif target_type == "integer":
-        y = pd.Series(np.random.randint(1, 25, 100))
+        y = [-1, -3, -5, 4, -2, 4, -4, 2, 1, 1]
     else:
-        y = np.random.normal(4, 1, 100)
+        y = [9.2, 7.66, 4.93, 3.29, 4.06, -1.28, 4.95, 6.77, 9.07, 7.67]
 
     y = infer_feature_types(y)
 
@@ -94,6 +92,8 @@ def test_target_distribution_data_check_warning_action(
         # This is essentially just checking the = of "shapiro_test_log.pvalue >= shapiro_test_og.pvalue"
         y = lognorm.rvs(s=1, loc=1, scale=1, size=10000)
 
+    y = np.round(y, 6)
+
     if data_type == "negative":
         y = -np.abs(y)
     elif data_type == "mixed":
@@ -114,7 +114,7 @@ def test_target_distribution_data_check_warning_action(
         shapiro_test_og = shapiro(y)
 
         details = {
-            "shapiro-statistic/pvalue": f"{round(shapiro_test_og.statistic, 3)}/{round(shapiro_test_og.pvalue, 3)}"
+            "shapiro-statistic/pvalue": f"{round(shapiro_test_og.statistic, 2)}/{round(shapiro_test_og.pvalue, 3)}"
         }
         assert target_dist_ == {
             "warnings": [

